src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -68,96 +52,154 @@ private Optional<EppoValue> getAssignmentValue(
+        if (rule.isEmpty()) {
+            log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
+            return null;
+        }
 
-        if (configuration.isBandit()) {
-            allocationKey = "bandit";
+        // Check if in experiment sample
+        String allocationKey = rule.get().getAllocationKey();
+        Allocation allocation = configuration.getAllocation(allocationKey);
+        int subjectShards = configuration.getSubjectShards();
+        if (!this.isInExperimentSample(subjectKey, flagKey, subjectShards,
+          allocation.getPercentExposure())) {
+            log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
+            return null;
+        }
 
-            // Properties of the bandit model are hardcoded for now
-            String modelName = "random";
-            String modelVersion = "0.1";
+        List<Variation> variations = allocation.getVariations();
 
-            assignmentModelVersion = modelName+" "+modelVersion;
+        String experimentKey = ExperimentHelper.generateKey(flagKey, allocationKey); // Used for logging
 
-            variations = BanditEvaluator.evaluateBanditVariations(
-              flagKey,
-              modelName,
-              assignmentOptions,
-              subjectKey,
-              subjectAttributes,
-              configuration.getSubjectShards()
-            );
-        } else {
-            // Find matched rule
-            Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
-            if (rule.isEmpty()) {
-                log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
-                return Optional.empty();
-            }
+        // Get assigned variation
+        String assignmentKey = "assignment-" + subjectKey + "-" + flagKey;
+        Variation assignedVariation = VariationHelper.selectVariation(assignmentKey, subjectShards, variations);
+
+        return new VariationAssignmentResult(
+          assignedVariation,
+          subjectKey,
+          subjectAttributes,
+          flagKey,
+          allocationKey,
+          experimentKey,
+          subjectShards
+        );
+    }
 
-            // Check if in experiment sample
-            allocationKey = rule.get().getAllocationKey();
-            Allocation allocation = configuration.getAllocation(allocationKey);
-            if (!this.isInExperimentSample(subjectKey, flagKey, configuration.getSubjectShards(),
-              allocation.getPercentExposure())) {
-                log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
-                return Optional.empty();
+    private Optional<EppoValue> determineAndLogBanditAction(VariationAssignmentResult assignmentResult, Map<String, EppoAttributes> assignmentOptions) {
```

Comment:
I know having the word "and" in a method name is a code smell ('it does two things!') but the theoretically cleaner way of having a separate `logBanditAction()` method would--as with variant selection--require passing around a ton of intermediate computed information and metadata. So then we'd need a `BanditActionResult` class. This dissuaded me from doing it, although it may indeed still be the better way.